### ISSUE TITLE
Notification "pop up" only show when app is visible

### ISSUE THREAD

**Issue by lukaskris** [MaikuB/flutter_local_notifications#657]
I using flutter_local_notification v. 1.4.4
My problem is notification show popup when the app is visible. butt on the background or when the phone on home screen notification doesn't show pop up, just vibrate.

this is a function for show notification
```
  static Future _showNotification(Map<String, dynamic> message) async {
    var pushTitle;
    var pushText;
    var action;

    if (Platform.isAndroid) {
      pushTitle = message['notification']['title'];
      pushText = message['notification']['body'];
      action = message['data']['order_header_id'];
    } else {
      pushTitle = message['title'];
      pushText = message['body'];
      action = message['action'];
    }

    // @formatter:off
    var androidPlatformChannelSpecifics = new AndroidNotificationDetails(
        Strings.notificationChannelId,
        Strings.notificationChannelName,
        Strings.notificationChannelDescription,
        playSound: true,
        enableVibration: true,
        importance: Importance.Max,
        priority: Priority.High);

    // @formatter:on
    var iOSPlatformChannelSpecifics =
        new IOSNotificationDetails(presentSound: false);
    var platformChannelSpecifics = new NotificationDetails(
        androidPlatformChannelSpecifics, iOSPlatformChannelSpecifics);

    await flutterLocalNotificationsPlugin.show(
      0,
      pushTitle,
      pushText,
      platformChannelSpecifics,
      payload: action,
    );
  }
```

this is for init local notification

```
_initLocalNotification() {
    var initializationSettingsAndroid =
        new AndroidInitializationSettings('@mipmap/ic_launcher');
    var initializationSettingsIOS = new IOSInitializationSettings();
    var initializationSettings = new InitializationSettings(
        initializationSettingsAndroid, initializationSettingsIOS);
    flutterLocalNotificationsPlugin.initialize(initializationSettings,
        onSelectNotification: onSelectNotification);
  }
```

this is for init firebase messenger
```
_initFirebaseMessaging() {
    _firebaseMessaging.configure(
      onMessage: (Map<String, dynamic> message) async {
        print("onMessage: $message");
        _showNotification(message);
      },
      onBackgroundMessage: Platform.isIOS ? null : myBackgroundMessageHandler,
      onLaunch: (Map<String, dynamic> message) async {
        print("onLaunch: $message");
        if (Platform.isIOS) {
          _showNotification(message);
        }
        return;
      },
      onResume: (Map<String, dynamic> message) async {
        print("onResume: $message");
        return;
      },
    );
    _firebaseMessaging.requestNotificationPermissions(
        const IosNotificationSettings(
            sound: true, badge: true, alert: true, provisional: true));
    _firebaseMessaging.onIosSettingsRegistered
        .listen((IosNotificationSettings settings) {
      print("Settings registered: $settings");
    });
    _firebaseMessaging.getToken().then((String token) async {
      assert(token != null);
      print("Push Messaging token: $token");
      _authenticationBloc.add(TokenUpdated(token));
    });
  }
```

**Comment by MaikuB** [comment]
This looks more to be an issue with `firebase_message` that your code would indicate is being used. Please make sure you look at the behaviour documented for `firebase_messaging`. Nothing you have there indicates that a you have isolated this to be issue with this plugin.

**Comment by lukaskris** [comment]
@MaikuB I've checked the documentation firebase. The background message was received on my device, but not show "pop up". There are screenshots when I stay on the app, and another screenshot I idle at home. From firebase, I think no problem cause can receive data. 

<img width="428" alt="image" src="https://user-images.githubusercontent.com/15793303/83787340-d8875900-a6bd-11ea-9a98-74f6d10bef45.png">
<img width="417" alt="image" src="https://user-images.githubusercontent.com/15793303/83787352-df15d080-a6bd-11ea-989e-11fb2e07199a.png">

**Comment by MaikuB** [comment]
Ok I see what you mean now. You're referring to a [heads-up notification](https://developer.android.com/guide/topics/ui/notifiers/notifications#Heads-up). The conditions for showing one is mentioned in that link. Generally this is to do with the importance. If you have the correct importance setup then it should appear *provided* that the code you have written is called.

> The background message was received on my device, but not show "pop up".

Have you actually confirmed that your code for showing a local notification is being called though? Note that you haven't provided the code for that either. I referred you to look at the `firebase_messaging` documentation as they describe what happens when the app is in the background on Android

> Notification is delivered to system tray. When the user clicks on it to open app onResume fires if click_action: FLUTTER_NOTIFICATION_CLICK is set (see below).

Note here that it mentions that `firebase_messaging` handles this. I even tried this on my app that doesn't have a background message handler by running the following as mentioned in their readme

```
DATA='{"notification": {"body": "this is a body","title": "this is a title"}, "priority": "high", "data": {"click_action": "FLUTTER_NOTIFICATION_CLICK", "id": "1", "status": "done"}, "to": "<FCM TOKEN>"}'
curl https://fcm.googleapis.com/fcm/send -H "Content-Type:application/json" -X POST -d "$DATA" -H "Authorization: key=<FCM SERVER KEY>"
```

I was able to the notification without having code for a background message handler. This doesn't have anything with `flutter_local_notifications`. As such, I'm closing this unless you're able to provide a complete standalone app and steps that shows that there's a bug with `flutter_local_notifications`
